src/tests/suite/daos_bigio.c

```diff
@@ -0,0 +1,505 @@
+/**
+ * (C) Copyright 2016 Intel Corporation.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ * GOVERNMENT LICENSE RIGHTS-OPEN SOURCE SOFTWARE
+ * The Government's rights to use, modify, reproduce, release, perform, display,
+ * or disclose this software are subject to the terms of the Apache License as
+ * provided in Contract No. B609815.
+ * Any reproduction of computer software, computer software documentation, or
+ * portions thereof marked with this legend must also reproduce the markings.
+ */
+/**
+ * This file is part of daos
+ *
+ * tests/suite/daos_bigio.c
+ */
+
+#include <daos/checksum.h>
+#include "daos_test.h"
+
+static void
+bigio_contig(void **state)
+{
+	test_arg_t	*arg = *state;
+	daos_obj_id_t	 oid;
+	daos_handle_t	 oh;
+	d_iov_t	 dkey;
+	d_sg_list_t	 sgl;
+	d_iov_t	 sg_iov;
+	daos_iod_t	 iod;
+	daos_recx_t	 recx;
+	int		*buf;
+	int		*buf_out;
+	daos_size_t	 i;
+	uuid_t		 co_uuid;
+	daos_cont_info_t co_info;
+	daos_handle_t	 coh;
+	int		 rc;
+
+	if (arg->myrank == 0) {
+		uuid_generate(co_uuid);
+		print_message("setup: creating container "DF_UUIDF"\n",
+			      DP_UUID(co_uuid));
+		rc = daos_cont_create(arg->pool.poh, co_uuid, NULL, NULL);
+		if (rc) {
+			print_message("daos_cont_create failed, rc: %d\n", rc);
+			goto bcast;
+		}
+
+		rc = daos_cont_open(arg->pool.poh, co_uuid, DAOS_COO_RW,
+				    &coh, &co_info, NULL);
+		if (rc)
+			print_message("daos_cont_open failed, rc: %d\n", rc);
+	}
+
+bcast:
+	if (arg->multi_rank)
+		MPI_Bcast(&rc, 1, MPI_INT, 0, MPI_COMM_WORLD);
+	if (rc)
+		assert_int_equal(rc, 0);
+
+	if (arg->multi_rank)
+		handle_share(&coh, HANDLE_CO, arg->myrank, arg->pool.poh, 0);
+
+	/*
+	 * Segfault in pmdk with 20GB I/O - 30 GB pool size.
+	 * This should be resolved though with breaking IOD at server side.
+	 */
+	arg->size = 4;
+	arg->nr = (daos_size_t)(1024*1024*1024)*20 / arg->size;
```

Comment:
this test will never work with the default pool size i believe.
also should be run over nvme most probably and not just tmpfs

Reply:
I am not sure whether run_tests.sh have basic infrastructure to format, prepare nvme now. Can we use tmpfs for now?

Reply:
given the nature of this test, this should really be done with NVMe for capacity.
i doubt this will all run successfully over just SCM.